src/Debugger/Impl/rtvs/R/grid.R

```diff
@@ -72,3 +72,53 @@ grid.dput <- function(obj) {
     json;
 }
 
+getEnvRepr <- function(env, higherThanGlobal) {
+  envRepr <- NULL;
+
+  if (higherThanGlobal) {
+    nframe <- sys.nframe();
+    if (nframe > 0) {
+      for (i in 1:nframe) {
+        frame <- sys.frame(i)
+        if (identical(env, frame)) {
+          envRepr <- list(name = deparse(sys.call(i)[[1]]), higherThanGlobal = TRUE);
+          break;
+        }
+      }
+    }
+  }
+
+  if (is.null(envRepr)) {
+    if (identical(env, baseenv())) {
+      envRepr <- list(name = 'package:base', higherThanGlobal = FALSE);
+    } else if (identical(env, globalenv())) {
+      envRepr <- list(name = '.GlobalEnv', higherThanGlobal = FALSE);
+    } else {
+      envRepr <- list(name = environmentName(env), higherThanGlobal = FALSE);
+    }
+  }
+  envRepr;
+}
+
+# return list of environment statck walking from the given environment up.
+getEnvironments <- function(env) {
```

Comment:
Do you plan to call this function for any environment, or only for those that are returned by `search()`?